framework/src/vectorpostprocessors/VectorPostprocessor.C

```diff
@@ -47,5 +48,8 @@ VectorPostprocessor::getVector(const std::string & vector_name)
 VectorPostprocessorValue &
 VectorPostprocessor::declareVector(const std::string & vector_name)
 {
-  return _vpp_fe_problem->declareVectorPostprocessorVector(_vpp_name, vector_name);
+  if (_vpp_tid)
+    return _thread_local_vector;
```

Comment:
Exactly as they are in UserObjects and Postprocessors. We already have a threadJoin() routine that's being called on these objects. The content of these extra vectors is more or less unimportant. It's just there as scratch space as the VectorPostprocessor executes.